pkg/chunk/cache/redis_client_test.go

```diff
@@ -0,0 +1,87 @@
+package cache
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/go-redis/redis/v8"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRedisClient(t *testing.T) {
+	single, err := mockRedisClientSingle()
+	require.Nil(t, err)
+	defer single.Close()
+
+	cluster, err := mockRedisClientCluster()
+	require.Nil(t, err)
+	defer cluster.Close()
+
+	ctx := context.Background()
+
+	clients := []*RedisClient{single, cluster}
+	for i, c := range clients {
+		meg := []string{"run on single redis client", "run on cluster redis client"}[i]
```

Comment:
It's a little convoluted. Instead of passing ` meg` (I guess you meant `msg`) to each assertion function, you can run the test within a `t.Run(description, func(t *testing.T) { /* your test logic here */})` and having the description only passed to `t.Run()`.

Following this example, a common pattern is using table testing to define the test cases. As example, I'm picking a random test where we use table testing:
https://github.com/cortexproject/cortex/blob/5009276fdd385587a13f1492520cc38647512bc6/pkg/chunk/schema_config_test.go#L306-L312

Reply:
thanks. fixed to assigned name and client to slice.